### PR TITLE
fix(codedeploy): ServerDeploymentGroup takes AutoScalingGroup instead of IAutoScalingGroup

### LINKED ISSUE #9175: ServerDeploymentGroup in Code deploy is accepting List of AutoScalingGroup instead of List of IAutoScalingGroup
ServerDeploymentGroup in Code deploy is accepting List of AutoScalingGroup instead of IAutoScalingGroup.

There is no way to get hold of reference of a AutoscallingGroup by using its name, rather `AutoScalingGroup.fromAutoScalingGroupName` gives an instance of IAutoScalingGroup, which can not be casted into AutoScalingGroup

**So, probably there is an oversight in ServerDeploymentGroup  construct. This should most likely accept List of IAutoScalingGroup instead of List of AutoScalingGroup . Kindly confirm.**

### PR DESCRIPTION
**Pull request by skinny85**
Fixes #9175

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts [hunk 1]
```diff
@@ -703,6 +710,7 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
 
     this.autoScalingGroup = new CfnAutoScalingGroup(this, 'ASG', asgProps);
     this.osType = imageConfig.osType;
+    this.operatingSystemType = imageConfig.osType;
```
Comment (NetaNir) [inline packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts:713]: Why do we need additional value? can't we use the `osType `? 
Should we add an enum value of `UNDEFINED` to the `OperatingSystemType ` Enum and keep one value?

Comment (skinny85) [reply packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts:713]: Sure. How about `UNKNOWN` for the third value?

Comment (NetaNir) [reply packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts:713]: sounds good

Comment (skinny85) [reply packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts:713]: @NetaNir what do you want to do for `OperatingSystemType.UNKNOWN` for this function: https://github.com/aws/aws-cdk/blob/b80c2718055a19a72955e457397d6e812a21e53e/packages/%40aws-cdk/aws-ec2/lib/user-data.ts#L88-L93 ?

**Pull request merged**